HoneyBadgers.RestApi/Repositories/IRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using HoneyBadgers.RestApi.Models;
+
+namespace HoneyBadgers.RestApi.Repositories
+{
+    public interface IRepository<T> where T: BaseEntity
+    {
+        IEnumerable<T> GetAll();
+        IQueryable<T> GetAllQueryable();
+        T Get(string id);
+        void Insert(T entity);
+        void Update(T entity);
+        void Delete(T entity);
```

Comment:
w kilku miejscach pojawia się to T entity, co, jak, dlaczego ;p?

Reply:
`T entity` to generyczne entity, pod `T` podstawiasz klasę i masz entity dalej klasy :) chyba lekko upraszczam ale mniej więcej o to chodzi

Reply:
jak w poprzednim komentarzu jedno repository dla wielu klas
więcej szczegółów [https://www.c-sharpcorner.com/article/generic-repository-pattern-in-asp-net-core/](tutaj https://www.c-sharpcorner.com/article/generic-repository-pattern-in-asp-net-core/)